src/cairo1-compiler.ts

```diff
@@ -9,53 +22,140 @@ export const exec = (args: string) => {
+            const isSameVersion =
+                exec(`${path.join(this.getBinDirPath(), "starknet-compile")}  --version`)
+                    .stdout.toString()
+                    .trim()
+                    .split(" ")[1] === this.compilerVersion;
+            if (isSameVersion) return this.getBinDirPath();
+        }
+
+        // Check machine type
+        const fileName = this.getOsSpecificFileName();
+        // Download compiler
+        await this.download(fileName);
+        await this.extractZipFile(fileName);
+        return this.getBinDirPath();
     }
 
-    async compileCairo1(): Promise<ProcessResult> {
-        const docker = await HardhatDocker.create();
-        if (!(await docker.hasPulledImage(this.image))) {
-            await docker.pullImage(this.image);
+    async download(fileName: string): Promise<void> {
+        const url = `${CAIRO_COMPILER_BINARY_URL}/v${this.compilerVersion}/${fileName}`;
+        try {
+            const response = await axios.get(url, {
+                responseType: "stream",
+                onDownloadProgress: (progressEvent) => {
+                    const totalLength = progressEvent.total;
+                    const downloadedLength = progressEvent.loaded;
+                    const percentage = Math.round((downloadedLength / totalLength) * 100);
+                    process.stdout.write(
+                        `Downloading cairo compiler version: ${this.compilerVersion} ... ${percentage}%\r`
+                    );
+                }
+            });
+
+            const destinationPath = path.join(this.compilerDownloadPath, fileName);
+
+            if (!fs.existsSync(this.compilerDownloadPath)) {
+                fs.mkdirSync(this.compilerDownloadPath, { recursive: true });
+            }
+
+            const writer = fs.createWriteStream(destinationPath);
+            response.data.pipe(writer);
+
+            await new Promise<void>((resolve, reject) => {
+                writer.on("finish", resolve);
+                writer.on("error", reject);
+            });
+
+            console.log("\nFile downloaded successfully!");
+        } catch (error) {
+            const parent = error instanceof Error && error;
+            throw new StarknetPluginError("Error downloading file:", parent);
         }
+    }
 
-        const { statusCode, stdout, stderr } = await docker.runContainer(
-            this.image,
-            this.getContainerArgs(),
-            {
-                binds: this.getDockerArgs()
+    async extractZipFile(fileName: FileName): Promise<void> {
+        try {
+            const zipFile = path.join(this.compilerDownloadPath, fileName);
+            const corelibPath = path.join(this.compilerDownloadPath, "target/corelib");
+            // Extract the tar/gz file
+            await tar.extract({
+                file: zipFile,
+                C: this.compilerDownloadPath,
+                strip: 1
```

Comment:
What do these two things mean?

Reply:
- `C` sets the directory where the extracted files will be placed
- `strip` removes a specified number of leading directory when extracting, in this case 1

Reply:
Can you add a comment in the code explaining why 1